Sources/Rendering/Core/ScalarBarActor/index.js

```diff
@@ -682,12 +685,29 @@ function vtkScalarBarActor(publicAPI, model) {
     mat4.transpose(cmat, cmat);
     mat4.invert(invmat, cmat);
 
-    const haveExtraColors =
-      typeof model.scalarsToColors.getNanColor === 'function' &&
+    let numberOfExtraColors = 0;
+    if (
+      model.drawNanAnnotation &&
+      typeof model.scalarsToColors.getNanColor === 'function'
+    ) {
+      numberOfExtraColors += 1;
+    }
+    if (
+      model.drawBelowRangeSwatch &&
+      typeof model.scalarsToColors.getBelowRangeColor === 'function' &&
+      model.scalarsToColors.getUseBelowRangeColor()
```

Comment:
strictly speaking, it is not because getBelowRangeColor is a function that get**Use**BelowRangeColor is a function.

I'm not sure about the difference between those 2 functions...

Reply:
`scalarsToColors` seems to be a vtk instance and you are tapping directly to methods from its publicAPI. To some extent, I'm not sure checking for type is needed as if present, the publicAPI "~only" have methods.